bootstrap/bootstrapper/fs/source.go

```diff
@@ -88,18 +71,25 @@ func (fss *fileSystemSource) GetAvailability(shard uint32, targetRangesForShard
 
 // ReadData returns raw series for a given shard within certain time ranges.
 func (fss *fileSystemSource) ReadData(shard uint32, tr xtime.Ranges) (m3db.ShardResult, xtime.Ranges) {
-	log := fss.opts.GetLogger()
-
 	if xtime.IsEmpty(tr) {
 		return nil, tr
 	}
-	files, err := fss.getInfoFiles(shard)
-	if err != nil {
-		log.Errorf("unable to get info files for shard %d: %v", shard, err)
+
+	var files []string
+	fs.ForEachInfoFile(
```

Comment:
Hey so yeah, I went back and forth on this but I really think we should have two methods:
```go
func InfoFiles() ([]string, error) { ... }
func ForEachInfoFile(fn func(fname string, data []byte)) { ... }
```

We should avoid getting into a "not that much overhead, let's just X" for as long as possible.